get.go

```diff
@@ -57,20 +61,95 @@ func Get(jwksURL string, options ...Options) (jwks *JWKs, err error) {
 // backgroundRefresh is meant to be a separate goroutine that will update the keys in a JWKs over a given interval of
 // time.
 func (j *JWKs) backgroundRefresh() {
+
+	// Create some rate limiting assets.
+	var lastRefresh time.Time
+	var queueOnce sync.Once
+	var refreshMux sync.Mutex
```

Comment:
I'm relatively happy with this custom rate limiter, but I'm not found of this mutex. It's mean to prevent race conditions surrounding the `lastRefresh` variable.

If there's a clean way around this, I'd like to consider it.